Sources/AblyChat/JSONCodable.swift

```diff
@@ -0,0 +1,9 @@
+internal protocol JSONEncodable {
+    var toJSONObjectValue: [String: JSONValue] { get }
+}
+
+internal protocol JSONDecodable {
+    init(jsonValue: JSONValue) throws
+}
+
+internal typealias JSONCodable = JSONDecodable & JSONEncodable
```

Comment:
Why couldn't we just use `Codable` as presence user data?

Reply:
The way you'd usually use `Codable` is with `JSONEncoder` which converts to and from `Data`. We want to convert to and from objects that can be passed to ably-cocoa such as `Dictionary` etc.